coins-on-the-clock/cpp/coins-on-the-clock.cpp

```diff
@@ -113,12 +113,7 @@ vector<string> getValidSequences(
     *(clockState + i) = false;
   }
 
-  // Initialize currentSequence to all 0s so we can
-  // I'm not actually sure I need to do this...
   int* currentSequence  = new int[numHours];
-  for (int i = 0; i < numHours; i += 1) {
```

Comment:
I'm confused as to whether or not you get zeroes automatically. My googling says that "new int[5]" is "default-initialized", while "new int[5]()" (note the parentheses) is "zero-initialized". You could simply add the parentheses and you should be okay.

Reply:
Oh, so I thought I would need this because I thought I was going to do something like
```
if currentSequence has numHours things in it {
  return valid sequence
else
...
```

but then I ended up having the separate `currentSequenceIndex` so I didn't need to worry about values in currentSequence.

Reply:
Also, it seems you may have lost some parentheses due to markdown.

In Markdown if you do \[thing\](otherthing) you embed a hyperlink with href "otherThing" and visible text "thing" so when you did \[5\]() it shows as a link with the text "5" that doesn't go anywhere :-).

Reply:
Wait a second, what's `default-initialized`? If my thing is an `int` array, then is the `default` already zero?

Reply:
My sources are below. Check the first answers of each.
1) https://stackoverflow.com/questions/2204176/how-to-initialise-memory-with-new-operator-in-c
2) https://stackoverflow.com/questions/10294801/c-initial-value-of-dynamic-array
3) https://stackoverflow.com/questions/7546620/operator-new-initializes-memory-to-zero

Reply:
Ah interesting so "default initialization" isn't "initialize to the default" it's "do the default thing regarding initialization (which in this case is 'do nothing')". I'm going to leave it as is because I don't need to 0 out the values and I take a minor performance hit by doing so, but that's really good to know and I think there's another place in this same file where I can do that instead (I think I make an array of bools that I manually default but could probably default this way)